src/uct/ib/ud/base/ud_iface.c

```diff
@@ -253,8 +253,15 @@ uct_ud_iface_create_qp(uct_ud_iface_t *self, const uct_ud_iface_config_t *config
     /* TODO: exp attrs autoconf */
     struct ibv_exp_qp_init_attr qp_init_attr;
     struct ibv_qp_attr qp_attr;
+    struct ibv_pd *pd;
     int ret;
 
+    pd = uct_ib_iface_md(&self->super)->pd;
```

Comment:
maybe create a short inline function for that, since it is used twice (at least)?